src/common/common.h

```diff
@@ -185,9 +196,33 @@ static __always_inline int file_info_from_ino(void *inode, file_info_t *file_inf
     if (read_field(inode, CRC_INODE_I_INO, &file_info->inode, sizeof(file_info->inode)) < 0)
         return -1;
 
+
+    // check sb magic to determine if the file in question is on a
+    // btrfs volume. if it is, the actual device ids that we care
+    // about are embedded in the btrfs_superblock struct.
+    unsigned long s_magic = 0;
+    if (read_field(i_sb, CRC_SBLOCK_S_MAGIC, &s_magic, sizeof(s_magic)) < 0) return -1;
+
+
     // device major/minor
     u32 i_dev = 0;
-    if (read_field(i_sb, CRC_SBLOCK_S_DEV, &i_dev, sizeof(i_dev)) < 0) return -1;
+    if (s_magic != BTRFS_SB_MAGIC) {
```

Comment:
nit: since we have an `else` I think I would rather have this `if` be for the case where it is equal (perhaps extracting the how to retrieve the inode into a function to make it shorter?) and the else the "default" case.